Bats test on server passed.

rubenchazarra opened the pull request:
Test on server passed. This PR includes some of the most used functions for scran 1.12. 
Some functions from scran 1.14 are also included as scripts, but outcomented in bats test. Will be updated soon.

rubenchazarra committed:
Add computeSumFactors script

rubenchazarra committed:
Scran scripts, test server

rubenchazarra committed:
Scran scripts, to test in server

pinin4fjords commented on dropletutils-read-10x-counts.R:
```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env Rscript 
```
Remember you don't need to duplicate this script. Use it from the dropletutils-scripts package :-)

pinin4fjords commented on get_experiment_data.R:
```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env Rscript 
```
This is Andrey's script, right? Again, use it from the Conda package.

Commit:
Error in read 10X data

Commit:
Testing scripts on server

Commit:
Bats test working

Commit:
Bats test working

rubenchazarra committed:
Bats test working

 On branch feature/initial_setups
 Your branch is ahead of 'origin/feature/initial_setups' by 3 commits.
   (use "git push" to publish your local commits)

 Changes to be committed:
   (use "git reset HEAD^1 <file>..." to unstage)

	deleted:    scater-calculate-qc-metrics.R
	deleted:    scater-filter.R

 Untracked files:
   (use "git add <file>..." to include in what will be committed)

	.scater-calculate-qc-metrics.R.swp
	.scran_post_install_tests.bats.swp
	post_install_tests/
	test_data/
	test_data_2/

rubenchazarra committed:
Uncomment export PATH=/nfs/production3/ma/home/rchazarra/scran-cli:/nfs/production3/ma/home/rchazarra/software/bin:/nfs/production3/ma/home/rchazarra/software/condabin:/usr/lib64/qt-3.3/bin:/ebi/lsf/ebi/ppm/10.2/bin:/ebi/lsf/ebi/ppm/10.2/linux2.6-glibc2.3-x86_64/bin:/ebi/lsf/ebi/10.1/linux3.10-glibc2.17-x86_64/etc:/ebi/lsf/ebi/10.1/linux3.10-glibc2.17-x86_64/bin:/usr/lpp/mmfs/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/bin:/usr/bin:/homes/rchazarra/bin

rubenchazarra committed:
Assessing 'get_experiment_data.R: command not found'

rubenchazarra committed:
Add artificial spike ins in ruben_filter_sce.rds

rubenchazarra committed:
Add artificial spike ins in ruben_filter_sce.rds

rubenchazarra committed:
Other envs scripts 'not found'

rubenchazarra committed:
get_experiment_data.R not found

rubenchazarra committed:
Remove bats from environment generation

rubenchazarra committed:
Remove bats from env and add Python 3.6

pinin4fjords committed:
Revert "Remove bats from env and add Python 3.6"

This reverts commit c9a22314218960182cde7a0e1fc155397c9ce9f7.

pinin4fjords committed:
Use smaller test set

rubenchazarra committed:
Retry python 3.6

pinin4fjords committed:
Temporary copy of get_experiment_data.R for testing

pinin4fjords committed:
Merge branch 'feature/initial_setups' of github.com:ebi-gene-expression-group/scran-cli into feature/initial_setups

pinin4fjords committed:
Revert "Retry python 3.6"

This reverts commit 8784b8495f160226889ffc5b3b3c7a165fee75b4.

pinin4fjords committed:
Just don't filter small matrices

pinin4fjords committed:
This script comes from external package

pinin4fjords committed:
Remove camel-case script names

pinin4fjords committed:
Pin packages

pinin4fjords committed:
Correct download method

pinin4fjords committed:
Testing: don't use import module

pinin4fjords committed:
Try some sleeps

pinin4fjords committed:
Use different sleep function

pinin4fjords committed:
Revert "Testing: don't use import module"

This reverts commit 68f088e1193039b82fc4e9dceeb316c76ccac538.

pinin4fjords committed:
Fix test downloader

pinin4fjords committed:
Dummy

pinin4fjords committed:
Sleeps don't work

pinin4fjords committed:
Remove test downloader script

a-solovyev12 commented on README.md:
```diff
@@ -1,2 +1,110 @@
+
+**Note**: this scripts are for Scran version 1.12. 
+
+## Install
```
@rubenchazarra,  you can add some notes on how to install the package even though it's not on Bioconda yet - something like `conda install scran-cli`. Also good to mention that it is better to install into a clean environment

a-solovyev12 commented on scran-compute-sum-factors.R:
```diff
@@ -0,0 +1,90 @@
+    action = "store",
+    default = "counts",
+    type = 'character',
+    help = 'Specify which assay values to use. Default: "logocunts'
```
@rubenchazarra, 'logocunts' is a nice one, but you'll have to replace it, I'm afraid ;)

a-solovyev12 commented on scran-cli-filter-sce.R:
```diff
@@ -0,0 +1,70 @@
+
+#add some genes as spike-ins, so that scran_computeSpikeFactors.R can run without error
+suppressPackageStartupMessages(require(SingleCellExperiment))
+sce <- `isSpike<-`(sce, type = "ERCC", value = c(sample(rownames(sce), 25)))
```
could you parameterise this line so that spike-ins are only added when specified by user?

rubenchazarra committed:
Add review tips

a-solovyev12 commented:
looks good to me - thanks @rubenchazarra
